op_hello/src/lib.rs

```diff
@@ -0,0 +1,41 @@
+extern crate deno;
+extern crate deno_std;
+use deno::*;
+
+pub fn init(&mut isolate: Isolate) -> Result<(), ErrBox> {
+  isolate.register_op("hello", op_hello); // register_op defined by #3002
+
+  // Explicitly link the deno_std crate so it can be used in hello_test.ts
+  // Its usage looks like this:
+  //
+  //   import { test } from "crate://deno_std/testing/mod.ts";
+  //
+  // In the future it might make sense to automate this function away, but I think
+  // it would be prudent to make the crate URL resolution as obvious as
+  // possible.
+  isolate.register_crate_url("deno_std", deno_std::get_file);
+
+  // TODO The ability to run typescript doesn't exist in deno core.
+  isolate.run("src/hello.ts")
```

Comment:
What about ES modules? AFAIK `Isolate` has only low level API for them and all the complexity is hidden in `modules.rs`.

Reply:
Yes we should support that too.